### ISSUE TITLE
Ideas for future work on Mods tab

### ISSUE THREAD

**Issue by Pidgeot** [Pidgeot/python-lnp#46]
If a manifest file is eventually added, it will be possible to do more with mods - and probably also make non-critical parts faster, but for most I value the correctness of file analysis over the speed of manifest reading.  I dislike the idea of a manifest, as it's a unique requirement which undermines interoperability and simplicity, and may not always be correct.

The big one is of course using mods for old DF versions - it would have to work by using the old mod, old vanilla, and new vanilla to create a new mod file (by the current 3-merge logic), and then use that in a mod merge for the new version.  

It would be possible to determine which DF version (of the available baselines) a mod is for, by checking how many lines of raws are changed relative to each and returning the least.  This would eliminate the requirement for a manifest, but force manual download of the relevant baseline and in some cases would slow the whole process a considerable amount.

If multiple baselines are present, they could be stored in reduced format relative to the newest.  I'm not sure if the uncommon disk space savings would be worth the runtime cost, but it's an idea.

TLDR; further enhancement is probably going to either force a manifest and thereby lose some elegance of the format, or may be too runtime-expensive to be worthwhile since the logic would have to run at least partially in many edge-ish cases such as multiple DF versions found, and could not store results (without... a manifest).

No action required, this just seemed a good place to dump my thinking that the mods tab is at a logical finishing point, and why.
